src/main/java/com/rabbitmq/http/client/domain/ShovelDetails.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.rabbitmq.http.client.domain;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+//{"src-uri":  "amqp://",              "src-queue":  "my-queue",
+//    "dest-uri": "amqp://remote-server", "dest-queue": "another-queue"}
+
+public class ShovelDetails {
+
+	@JsonProperty("src-uri")
+	private String sourceURI;
+	@JsonProperty("src-exchange")
+	private String sourceExchange;
+	@JsonProperty("src-queue")
+	private String sourceQueue;
+
+	@JsonProperty("dest-uri")
+	private String destURI;
+	@JsonProperty("dest-exchange")
+	private String destExchange;
+	@JsonProperty("dest-queue")
+	private String destQueue;
+
+	@JsonProperty("reconnect-delay")
+	private long reconnectDelay;
+	@JsonProperty("add-forward-headers")
+	private boolean addForwardHeaders;
+
+	@JsonProperty("publish-properties")
+	private Map<String, Object> publishProperties;
+
+	public ShovelDetails() {
+	}
+
+	public ShovelDetails(String sourceURI, String destURI, long reconnectDelay, boolean addForwardHeaders, Map<String, Object> publishProperties) {
+		this.sourceURI = sourceURI;
+		this.destURI = destURI;
+		this.reconnectDelay = reconnectDelay;
+		this.addForwardHeaders = addForwardHeaders;
+		this.publishProperties = publishProperties;
+	}
+
+	public String getSourceURI() {
+		return sourceURI;
+	}
+
+	public void setSourceURI(String sourceURI) {
+		this.sourceURI = sourceURI;
+	}
+
+	public String getSourceExchange() {
+		return sourceExchange;
+	}
+
+	public void setSourceExchange(String sourceExchange) {
+		this.sourceExchange = sourceExchange;
+	}
+
+	public String getSourceQueue() {
+		return sourceQueue;
+	}
+
+	public void setSourceQueue(String sourceQueue) {
+		this.sourceQueue = sourceQueue;
+	}
+
+	public String getDestURI() {
```

Comment:
Thank you. Can we please use full words in field and accessor names? Yes, the API ones are abbreviated and can't change but we can do better here :)